scanpy/tests/test_highly_variable_genes.py

```diff
@@ -338,23 +338,26 @@ def test_pearson_residuals_batch(pbmc3k_parametrized_small, subset, n_top_genes)
         ),
     ],
 )
-def test_compare_to_upstream(
+@pytest.mark.parametrize("array_type", ARRAY_TYPES)
+def test_compare_to_upstream(  # noqa: PLR0917
     request: pytest.FixtureRequest,
     func: Literal["hvg", "fgd"],
     flavor: Literal["seurat", "cell_ranger"],
     params: dict[str, float | int],
     ref_path: Path,
+    array_type: list,
```

Comment:
also no. 1. never use `list` as a parameter type (only return type) and 2. always parameterize collection types